packages/frontapp/src/app/components/Scanner/index.tsx

```diff
@@ -0,0 +1,307 @@
+      border-right: 3px solid ${colors.bg_secondary};
+      border-top-right-radius: ${dimensions.xl};
+    }
+
+    &.bl {
+      bottom: 0;
+      left: 0;
+      border-bottom: 3px solid ${colors.bg_secondary};
+      border-left: 3px solid ${colors.bg_secondary};
+      border-bottom-left-radius: ${dimensions.xl};
+    }
+
+    &.br {
+      bottom: 0;
+      right: 0;
+      border-bottom: 3px solid ${colors.bg_secondary};
+      border-right: 3px solid ${colors.bg_secondary};
+      border-bottom-right-radius: ${dimensions.xl};
+    }
+  }
+`;
+
+const ScannerCanvas = styled.canvas`
+  display: none;
+  width: 100%;
+  margin: auto;
+`;
+
+const ScannerImage = styled.img`
+  display: none;
+  width: 100%;
+  margin: auto;
+`;
+
+const CloseButton = styled.div`
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  position: absolute;
+  top: ${dimensions.base};
+  right: ${dimensions.base};
+  width: ${dimensions.xl_6};
+  height: ${dimensions.xl_6};
+  background-color: ${colors.bg_secondary};
+  border-radius: 50%;
+
+  svg {
+    fill: ${colors.accent_color};
+    width: ${dimensions.xl_3};
+    height: ${dimensions.xl_3};
+  }
+`;
+
+const Scanner: FC<IScannerProps> = ({ onDetected, onClose }) => {
+  let videoStream: MediaStream | null;
+  const scannerElement = useMemo(() => document.querySelector('#scanner')!, []);
+  const barcodeReader = new BrowserMultiFormatReader();
+  const timeout = 1000; // time between frames
+
+  useEffect(() => {
+    showScanner();
+
+    const videoElement = document.querySelector<HTMLVideoElement>(
+      '#scanner-video video'
+    )!;
+    const canvasElement =
+      document.querySelector<HTMLCanvasElement>('#scanner-canvas')!;
+    const imageElement =
+      document.querySelector<HTMLImageElement>('#scanner-image')!;
+    const frameElement =
+      document.querySelector<HTMLDivElement>('#scanner-frame')!;
+
+    const constraints = { video: true };
+    const frameSize = dinamicFrameSize(window.innerWidth, window.innerHeight);
+
+    navigator.mediaDevices.getUserMedia(constraints).then((stream) => {
+      videoStream = stream;
+
+      // handle play callback
+      videoElement.addEventListener('play', () => {
+        // get video's intrinsic width and height, eg 640x480,
+        // and set canvas to it to match.
+
+        canvasElement.width = frameSize.width;
+        canvasElement.height = frameSize.height;
+
+        // set position of orange frame in video
+        frameElement.style.width = `${frameSize.width}px`;
+        frameElement.style.height = `${frameSize.height}px`;
+        frameElement.style.left = `${
+          (window.innerWidth - frameSize.width) / 2
+        }px`;
+        frameElement.style.top = `${
+          (window.innerHeight - frameSize.height) / 2
+        }px`;
+
+        // start the barcode reader process
+        scanFrame();
+      });
+
+      videoElement.srcObject = stream;
+    });
+
+    function scanFrame() {
```

Comment:
why not `const scanFrame = ()=>{}` ?

Reply:
I think it's not very important. Here I preferred FD instead of FE